### DISCUSSION TITLE
remix.config.js publicPath outside `public`

### DISCUSSION THREAD

**Discussion by akomm** [remix-run/remix#1081] [Q&A]
Given:
```js
{
  browserBuildDirectory: ".build/browser",
  assetsBuildDirectory: ".build/browser",
  publicPath: "/",
  serverBuildDirectory: ".build/server",
}
```
What does not make any sense: 
`browserBuildDirectory` needs to be relative to `remix.config.js`, even though it looks like `publicPath` is hard-coded to be relative to `./public/`. So setting `browserBuildDirectory` to anything outside of `public/` does currently not make any sense.

I'v tried dozens of combinations to "guess" how it might work, but only those inside `public/` worked, everything outside ended up in 404s.

From my technical understanding it should not be a problem to serve static files from any document root/directory, isolating it from server files lie in my example.

**Comment by ryanflorence** [comment]
`publicPath` is what remix uses to build the URLs from the browser's perspective.

It sounds like you're expecting this to **configure your server** to serve files from the directory you specify. Remix is not your server, so this isn't a server setting, you own that with your server. This setting *instructs remix of your server setting* when it generates URLs to assets.

**Comment by akomm** [comment]
@ryanflorence 
Thanks. I know its just functions, I should have added: using the remix app server and `run dev`.
